app/com/linkedin/drelephant/exceptions/azkaban/AzkabanJobLogExceptionFingerprinting.java

```diff
@@ -0,0 +1,85 @@
+package com.linkedin.drelephant.exceptions.azkaban;
+
+import com.linkedin.drelephant.clients.WorkflowClient;
+import com.linkedin.drelephant.configurations.scheduler.SchedulerConfigurationData;
+import com.linkedin.drelephant.exceptions.util.ExceptionInfo;
+import com.linkedin.drelephant.util.InfoExtractor;
+import com.linkedin.drelephant.util.Utils;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.log4j.Logger;
+
+import static com.linkedin.drelephant.exceptions.util.ExceptionUtils.ConfigurationBuilder.*;
+import static com.linkedin.drelephant.exceptions.util.ExceptionUtils.*;
+import static controllers.api.v1.JsonKeys.*;
+
+
+public class AzkabanJobLogExceptionFingerprinting {
+
+  private WorkflowClient _workflowClient;
+  private List<ExceptionInfo> azkabanExceptionInfoList = new ArrayList<>();
+  private final String SCHEDULER_NAME = "azkaban";
+  private final String PRIVATE_KEY = "private_key";
+
+  private static final Logger logger = Logger.getLogger(AzkabanJobLogExceptionFingerprinting.class);
+
+  public  AzkabanJobLogExceptionFingerprinting(String flowExecUrl, String jobExecUrl) {
+    logger.info("Initailizing AzkabanExceptionFingerPrinting class for " + jobExecUrl);
+    String jobId = getJobName(jobExecUrl);
+    _workflowClient = getWorkflowClient(flowExecUrl);
+    logger.info("Fetching job logs from Azkaban for " + jobExecUrl);
+    String jobLog = _workflowClient.getAzkabanJobLog(jobId, AZKABAN_JOB_LOG_START_OFFSET.getValue()
+            .toString(), AZKABAN_JOB_LOG_MAX_LENGTH.getValue().toString());
+    analyzeJobLog(jobLog);
+  }
+
+  public List<ExceptionInfo> getExceptionInfoList() {
+    return azkabanExceptionInfoList;
+  }
+
+  private void analyzeJobLog(String jobLogs) {
+    logger.info("Analyzing job logs from Azkaban");
```

Comment:
The offset can be set from properties. Since Azkaban don't provide any API to fetch log length information, we need to fetch whole logs irrespective of size.

Reply:
Whats the response time expectations if the Logs are in 100MB  , Have we seen cases where logs are hundreds of MB.  Even if we got the complete log , we should do the parsing of last some bytes only

Reply:
I don't have any such benchmark data present as of now.

Reply:
can we put timer now to know (in logs) , how much time they are taking and put it

Reply:
Sure